provider/cmd/pulumi-resource-mongodbatlas/init/init.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2016-2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package initialize
+
+import "os"
+
+func init() {
+	// This variable affects the schema of the provider. The most foolproof way to make sure it
+	// is set in local and CI build contexts is setting it in code over here.
+	os.Setenv("MONGODB_ATLAS_ENABLE_PREVIEW", "true")
```

Comment:
Do we need to cleanup and unset/revert the env var when provider shuts down?

Reply:
If I understand env vars correctly, this only affects the provider process itself and any sub-processes it may spawn (this one doesn't spawn afaik). No process can modify the caller's environment AFAIK. So no.